docs/docs/integrations/providers/cratedb.mdx

```diff
@@ -0,0 +1,203 @@
+how to use it for similarity search and other purposes.
+
+
+## What is CrateDB?
+
+[CrateDB] is an open-source, distributed, and scalable SQL analytics database
+for storing and analyzing massive amounts of data in near real-time, even with
+complex queries. It is PostgreSQL-compatible, based on [Lucene], and inherits
+the shared-nothing distribution layer of [Elasticsearch].
+
+It provides a distributed, multi-tenant-capable relational database and search
+engine with HTTP and PostgreSQL interfaces, and schema-free objects. It supports
+sharding, partitioning, and replication out of the box.
+
+CrateDB enables you to efficiently store billions of records, and terabytes of
+data, and query it using SQL.
+
+- Provides a standards-based SQL interface for querying relational data, nested
+  documents, geospatial constraints, and vector embeddings at the same time.
+- Improves your operations by storing time-series data, relational metadata,
+  and vector embeddings within a single database.
+- Builds upon approved technologies from Lucene and Elasticsearch.
+
+
+## CrateDB Cloud
+
+- Offers on-demand CrateDB clusters without operational overhead, 
+  with enterprise-grade features and [ISO 27001] certification.
+- The entrypoint to [CrateDB Cloud] is the [CrateDB Cloud Console].
+- Crate.io offers a free tier via [CrateDB Cloud CRFREE].
+- To get started, [sign up] to CrateDB Cloud, deploy a database cluster,
+  and follow the upcoming instructions.
+
+
+## Features
+
+The CrateDB adapter supports the _Vector Store_, _Document Loader_,
+and _Conversational Memory_ subsystems of LangChain.
+
+### Vector Store
+
+`CrateDBVectorSearch` is an API wrapper around CrateDB's `FLOAT_VECTOR` type
+and the corresponding `KNN_MATCH` function, based on SQLAlchemy and CrateDB's
+SQLAlchemy dialect. It provides an interface to store and retrieve floating
+point vectors, and to conduct similarity searches.
+
+Supports:
+- Approximate nearest neighbor search.
+- Euclidean distance.
+
+### Document Loader
+
+`CrateDBLoader` provides loading documents from a database table by an SQL
+query expression or an SQLAlchemy selectable instance.
+
+### Conversational Memory
+
+`CrateDBChatMessageHistory` uses CrateDB to manage conversation history.
+
+
+## Installation and Setup
+
+There are multiple ways to get started with CrateDB.
+
+### Install CrateDB on your local machine
+
+You can [download CrateDB], or use the [OCI image] to run CrateDB on Docker or Podman.
+Note that this is not recommended for production use.
+
+```shell
+docker run --rm -it --name=cratedb --publish=4200:4200 --publish=5432:5432 \
+    --env=CRATE_HEAP_SIZE=4g crate/crate:nightly \
+    -Cdiscovery.type=single-node
+```
+
+### Deploy a cluster on CrateDB Cloud
+
+[CrateDB Cloud] is a managed CrateDB service. Sign up for a [free trial].
+
+### Install Client
+
+```bash
+pip install 'crate[sqlalchemy]' 'langchain[openai]' 'crash'
```

Comment:
That installation incantation will be deprecated. Something like this would be better:

```suggestion
pip install crash langchain langchain-openai sqlalchemy-cratedb
```